consensus/abci.go

```diff
@@ -123,17 +122,50 @@ func (abci *TendermintABCI) CheckTx(txBytes []byte) tmtAbciTypes.ResponseCheckTx
 	
 	tx, err := decodeRLP(txBytes)
 	if err != nil {
-		abci.logger.Error("Received invalid transaction", "tx", tx.Hash().String())
-		abci.metrics.CheckErrTxsTotal.Add(1, fmt.Sprint(abciTypes.ErrEncodingError.Code))
-		return tmtAbciTypes.ResponseCheckTx{Code: uint32(abciTypes.ErrEncodingError.Code), Log: err.Error()}
+		abci.logger.Error("Unable to decode RLP TX", "err", err.Error())
+		abci.metrics.CheckErrTxsTotal.Add(1, "INVALID_TX")
+		return tmtAbciTypes.ResponseCheckTx{Code: 1, Log: "INVALID_TX"}
 	}
 
 	abci.logger.Info("Checking TX", "hash", tx.Hash().String(), "nonce", tx.Nonce(), "cost", tx.Cost())
+	from, err := abci.doMempoolValidation(tx)
```

Comment:
[MAJOR] Why a `doMempoolValidation` returns an error along with a Eth account. That is hard to understand an it breaks the Single Responsibility propose of this method.

Reply:
because it was convenient and I was in rush but yea I agree, PS: [MINOR]